proto/streamlit/proto/Delta.proto

```diff
@@ -26,12 +26,19 @@ message Delta {
     Element new_element = 3;
 
     // Append a new block to the report.
-    bool new_block = 4;
+    Block add_block = 6;
 
     // Append data to a DataFrame in for current element. The element to add to
     // is identified by the ID field, above. The dataframe is identified either
     // by NamedDataSet.name or by setting NamedDataSet.has_name to false.
     // All elements that contain a DataFrame should support add_rows.
     NamedDataSet add_rows = 5;
   }
+
+  message Block {
```

Comment:
I don't think the final proto structure is going to be much like this, but I also think that's fine; we can merge this PR as is, then when we add in columns I'll just overhaul it.

FWIW I'm thinking something more self-contained and extensible like:
```
  message Block {
    oneof type {
      Vertical vertical = 1;
      Collapsible collapsible = 2;
      Horizontal horizontal = 3;
      Column column = 4;
    }

    message Vertical {
      bool unused = 1;
    }

    message Collapsible {
      string label = 1;
      bool collapsed = 2;
    }

    message Horizontal { 
      bool unused = 1;
    }

    message Column {
      double weight = 1;
    }
  }
```

Reply:
I do like this, making collapsible into its own message. I think I'll do that now. Not sure if this is the right place but is it weird to have two separate messages for horizontal and vertical? Could you have a block that is both?

Reply:
Mm, the reason to have horiz/vertical is to disambiguate them. We could use a single message with a isHorizontal bool, but maybe in the future vertical columns have a "width" field" while horizontal ones have a "height"; might as well conceptually split these up so every different kind of block gets its own area.